internal/scorecard/storage.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubectl/pkg/scheme"
+)
+
+const (
+	StorageSidecarContainer = "scorecard-gather"
+)
+
+func (r PodTestRunner) execInPod(podName, mountPath, containerName string) (io.Reader, error) {
+	cmd := []string{
+		"tar",
+		"cf",
+		"-",
+		mountPath,
+	}
+
+	reader, outStream := io.Pipe()
+	const tty = false
+	req := r.Client.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(podName).
+		Namespace(r.Namespace).SubResource("exec").Param("container", containerName)
+	req.VersionedParams(
+		&v1.PodExecOptions{
+			Command: cmd,
+			Stdin:   false,
+			Stdout:  true,
+			Stderr:  true,
+			TTY:     tty,
+		},
+		scheme.ParameterCodec,
+	)
+
+	var stderr bytes.Buffer
+	exec, err := remotecommand.NewSPDYExecutor(r.RESTConfig, "POST", req.URL())
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		defer outStream.Close()
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  nil,
+			Stdout: outStream,
+			Stderr: &stderr,
```

Comment:
stderr gets thrown away here I think. This should be returned and set in test output if not empty.

Reply:
great catch